apps/backend/app/services/member_stats_service.py

```
@@ -0,0 +1,244 @@
+                "$min": {"first_sent_at": at},
+                "$push": {"touches": {"$each": [touch], "$slice": -TOUCH_HISTORY_LIMIT}},
+                "$setOnInsert": {
+                    "restaurant_id": restaurant_id,
+                    "phone_key": key,
+                    "received_count": 0,
+                    "read_count": 0,
+                },
+            },
+            upsert=True,
+        )
+    except Exception as e:  # never let stats bookkeeping break a send
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)